YamlDotNet/Core/ParserExtensions.cs

```diff
@@ -147,5 +147,63 @@ public static void SkipThisAndNestedEvents(this IParser parser)
         {
             return Accept<T>(parser, out var _);
         }
+
+        /// <summary>
+        /// Checks whether the current event contains a mapping entry that returns true for the specified selector.
+        /// If one is found, it's key is returned as a Scalar and it's value as a ParsingEvent, the mapping entry
+        /// is consumed and the parser is moved to the next event, and the whole function returns true.
+        /// If the event is not a mapping event or a matching key is not found, returns false.
+        /// </summary>
```

Comment:
```suggestion
         /// Attempts to find a key on a YAML mapping that matches our predicate.
         /// This is useful for scanning a mapping for type discriminator information.
         /// For example: looking for a `kind` key on an object.
         ///
         /// This function only checks mappings, and only looks at the current depth.
         ///
         /// If the event is a mapping and has a key that satisfies the predicate the scan 
         /// will stop, return true, and set
         /// <paramref>key</paramref> and <paramref>value</paramref>.
         ///
        /// If the event is not a mapping event or a matching key is not found, returns false.
```

Your description is correct but perhaps a little jargony. I've added my suggestion, feel free to adapt how you like.